src/no-random-v4-uuid.spec.ts

```diff
@@ -70,10 +80,166 @@ createTester().run(ruleId, rule, {
                const uuidV4 = v4();
             
                const uuidV5 = v5('name', namespace);`,
+      errors: [
+        {
+          messageId: 'NO_UUID_MODULE_FOR_V4',
+        },
+        {
+          messageId: 'NO_UUID_MODULE_FOR_V4',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with crypto.randomUUID',
+      code: `import { randomUUID } from 'node:crypto';
+         const id = randomUUID();`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with alias import from crypto',
+      code: `import { randomUUID as uuid } from 'node:crypto';
+         const id = uuid();`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with all the named exports crypto namespace',
+      code: `import * as crypto from 'node:crypto';
+         const id = crypto.randomUUID();`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with mixed imports',
+      code: `import { randomUUID, randomBytes } from 'node:crypto';
+         const id = randomUUID();
+         const bytes = randomBytes(16);`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with crypto namespace',
+      code: `import crypto from 'node:crypto';
+         const id = crypto.randomUUID();`,
       errors: [
         {
           messageId: 'NO_RANDOM_V4_UUID',
         },
+      ],
+    },
+    {
+      name: 'Invalid case with mixed import and alias',
+      code: `import crypto, { randomInt, randomUUID as uuid } from 'node:crypto';
+         const id = uuid();`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with uuid module used only for v4 UUID generation',
+      code: `import { v4 } from 'uuid';
+         const id = v4();`,
+      errors: [
+        {
+          messageId: 'NO_UUID_MODULE_FOR_V4',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with uuid module used only for alias v4 UUID generation',
+      code: `import { v4 as uuid } from 'uuid';
+         const id = uuid();`,
+      errors: [
+        {
+          messageId: 'NO_UUID_MODULE_FOR_V4',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with standard JS crypto.randomUUID',
+      code: `const id = crypto.randomUUID();`,
+      errors: [
+        {
+          messageId: 'NO_RANDOM_V4_UUID',
+        },
+      ],
+    },
+    {
+      name: 'Invalid case with standard JS crypto.randomUUID in a function',
```

Comment:
it's probably just me, this and the following tests for testing different surrounding structures seem unnecessary because we know the linting logic focuses only on the call expression